### ISSUE TITLE
Remove `apply_softmax` in cross entropy loss class

### ISSUE THREAD

**Issue by jhossbach** [zincware/ZnNL#70]
We use `optax.softmax_cross_entropy` for the cross entropy calculation which expects unnormalized log probabilities, so we would be applying the softmax twice if we initialize our cross entropy loss with `apply_softmax=True`
